Sources/PaymentsCore/Networking/APIClient.swift

```diff
@@ -4,22 +4,42 @@ import Foundation
 ///
 /// `APIClient` is the entry point for each payment method feature to perform API requests. It also offers convenience methods for API requests used across multiple payment methods / modules.
 public class APIClient {
+        
+    // MARK: - Internal Properties
 
     private var http: HTTP
     private let sessionID = UUID().uuidString.replacingOccurrences(of: "-", with: "")
     private let coreConfig: CoreConfig
+    
+    /// The `AnalyticsService` instance is static/shared so that only one sessionID is used.
+    /// The "singleton" has to be managed here because `AnalyticsService` has a dependency on `HTTP`.
+    ///
+    /// Exposed for testing.
+    weak var analyticsService: AnalyticsService? {
+        get { APIClient._analyticsService }
+        set { APIClient._analyticsService = newValue }
+    }
+    private static var _analyticsService: AnalyticsService?
+    
+    // MARK: - Public Initializer
 
     public init(coreConfig: CoreConfig) {
         self.http = HTTP(coreConfig: coreConfig)
         self.coreConfig = coreConfig
+        APIClient._analyticsService = AnalyticsService(http: http)
```

Comment:
q: should this create a new `AnalyticsService` every time an `APIClient` is created?